pkg/loghttp/params.go

```diff
@@ -73,8 +73,8 @@ func step(r *http.Request, start, end time.Time) (time.Duration, error) {
 
 // defaultQueryRangeStep returns the default step used in the query range API,
 // which is dinamically calculated based on the time range
-func defaultQueryRangeStep(start time.Time, end time.Time) int {
-	return int(math.Max(math.Floor(end.Sub(start).Seconds()/250), 1))
+func defaultQueryRangeStep(start time.Time, end time.Time) time.Duration {
+	return end.Sub(start) / 250
```

Comment:
Looks like the / 250 was applied to seconds. Now it's applied to nanoseconds. Doesn't that changes the behaviour ?

/cc @pracucci WDYT ? You did that piece.

Reply:
Yes, the current behaviour changes as can be seen on the tests. Now the range is divided in 250 precise steps. For example, in the previous implementation if you had a range of 1h you would get steps of 14s, and now you would get steps of 14,4s.

Reply:
Ha that makes sense ! since you returns nanoseconds now.

Reply:
Also the value is not capped to 1s anymore. So if you have a range of, let's say, 200seconds you will get steps of 0.8s. I think it makes sense, but maybe someone else with more knowledge on this part will know better.

Reply:
The change itself doesn't look wrong, but I'm not sure this change brings any real benefit and I'm not getting the reason why we should support steps below 1s. Which use case do we want to cover?

Reply:
You've got a good point there. There's no real use case at the moment that supports this change. It was just an idea we had with Cyril after pushing this change: https://github.com/grafana/loki/pull/1355

Actually, thinking out loud Prometheus doesn't seem to provide a default value for this parameter, so if we want Loki to have an API matching Prometheus' API, we should remove this feature.